opensrp-chw/src/kk/java/org/smartregister/chw/actionhelper/NewBornCareBreastfeedingHelper.java

```diff
@@ -0,0 +1,128 @@
+        try {
+            JSONObject jsonObject = new JSONObject(jsonString);
+            JSONArray fields = JsonFormUtils.fields(jsonObject);
+
+            if (serviceWrapperMap != null) {
+                String serviceName = serviceWrapperMap.getName();
+                if ("months".equalsIgnoreCase(getPeriodNoun(serviceName))) {
+                    if (getPeriod(serviceName) > 5) {
+                        JsonFormUtils.getFieldJSONObject(fields, "visit_months").put(JsonFormConstants.VALUE, "yes");
+                    }
+                }
+            }
+
+            if (firstBreastFeedingVisitHappened) {
+                JsonFormUtils.getFieldJSONObject(fields, "first_visit").put(JsonFormConstants.VALUE, "visit_done");
+            }
+
+            return jsonObject.toString();
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+        return super.getPreProcessed();
+    }
+
+    @Override
+    public void onPayloadReceived(String jsonPayload) {
+        try {
+            JSONObject jsonObject = new JSONObject(jsonPayload);
+            breastfed_prev = org.smartregister.chw.util.JsonFormUtils.getValue(jsonObject, "breastfed_prev");
+            time_to_breastfeed = org.smartregister.chw.util.JsonFormUtils.getValue(jsonObject, "time_to_breastfeed");
+            breastfeed_current = org.smartregister.chw.util.JsonFormUtils.getValue(jsonObject, "breastfeed_current");
+            other_food_child_feeds = org.smartregister.chw.util.JsonFormUtils.getValue(jsonObject, "other_food_child_feeds");
+        } catch (Exception e) {
+            Timber.e(e);
+        }
+
+    }
+
+    @Override
+    public String evaluateSubTitle() {
+        return null;
+    }
+
+    @Override
+    public BaseAncHomeVisitAction.Status evaluateStatusOnPayload() {
+        BaseAncHomeVisitAction.Status status;
+        if (firstBreastFeedingVisitHappened) {
+            if (StringUtils.isBlank(breastfeed_current) || StringUtils.isBlank(other_food_child_feeds)) {
+                status = BaseAncHomeVisitAction.Status.PARTIALLY_COMPLETED;
+            } else {
+                status = BaseAncHomeVisitAction.Status.COMPLETED;
+            }
+        } else {
+            if (StringUtils.isBlank(breastfed_prev) || StringUtils.isBlank(time_to_breastfeed) ||
+                    StringUtils.isBlank(breastfeed_current) || StringUtils.isBlank(other_food_child_feeds)) {
+                status = BaseAncHomeVisitAction.Status.PARTIALLY_COMPLETED;
+            } else {
+                status = BaseAncHomeVisitAction.Status.COMPLETED;
+            }
+        }
+        return status;
+    }
+
+    private String getPeriodNoun(String serviceName) {
+        String[] nameSplit = serviceName.split(" ");
+        return nameSplit[nameSplit.length -1];
+    }
+
+    private int getPeriod(String serviceName) {
+        String[] nameSplit = serviceName.split(" ");
+        String periodString = nameSplit[nameSplit.length -2];
```

Comment:
Same comment, will month 9 and month 10 return the same?

Reply:
Yes @issyzac - It spits the spaces, so New Breastfeeding 12 months when spited will be ["New", "Breastfeeding", "12", "Month"]. Which should be the same when it is 9 months